### DISCUSSION TITLE
PrimeVue - Datatable - Column how to add a mouseover and mouseleave event to whole column

### DISCUSSION THREAD

**Discussion by barrierwill** [primefaces/community#861] [PrimeVue] [answered]
Version: 3.42.0
Problem: I am trying to add a hover affect to my table that changes the color of the column on the table that the user is currently hovering over. My thought process was to add a mouseover / mouseleave event to my column that updates a value that keeps track of the user hovering over the column to add the styles needed. This is how I thought it could work: 
```
<ColumnPrimeVue field="inBasicPlan" style="width:20%"
                @mouseover="basicHover = true" @mouseleave="basicHover = false">
    
</ColumnPrimeVue>
```

However, this does nothing. There is no mouseover/mouseleave event that occurs. I instead have to do something like this: 
```
<ColumnPrimeVue field="inBasicPlan" style="width:20%">
    <template #header>
        <div  @click="selectPlanName('Plan 1')" class="text-center" @mouseover="basicHover = true" @mouseleave="basicHover = false" >
            Plan 1
        </div>
    </template>

    <template #body="{data}">
        <div @click="selectPlanName('Plan 1')" class="text-center" @mouseover="basicHover = true" @mouseleave="basicHover = false" >
           {{data.feature}}
        </div>
    </template>
</ColumnPrimeVue>
```
This is leading to some unintended results while the user hovers because the mouseover / mouseleave events are only applied to the div inside of the cell. It is not applied to the td element itself. It makes sense why the td element doesn't have the mouseover / mouseleave event because these events are applied to the divs in the template. Is there a way to pass the mouse events directly to the td?

Thank you for your help.

**Comment by mertsincan** [comment, marked as answer]
Hi,

You can use `pt` with `bodycell` key; 
```
<DataTable/Column :pt="{ 
         bodyCell: {
               onMouseover: (e) => ...,
               onMouseleave: (e) => ...,
         }
}"
```
